Build/Build.cs

```diff
@@ -305,19 +311,60 @@ from framework in frameworks
                             $"--report-trx-filename {v.project.Name}_{v.framework}.trx",
                             $"--results-directory {TestResultsDirectory}"
                         )
-                    )
-                );
+                )
+            );
         });
 
     Target TestFrameworks => _ => _
         .DependsOn(VSTestFrameworks)
         .DependsOn(TestingPlatformFrameworks);
 
+    Target SonarBegin => _ => _
+        .Before(Compile)
+        .Executes(() =>
+        {
+            SonarScannerTasks.SonarScannerBegin(s =>
+            {
+                SonarScannerBeginSettings settings = s
+                    .SetFramework("net8.0")
+                    .SetProjectKey("AwesomeAssertions_AwesomeAssertions")
+                    .SetOrganization("awesomeassertions")
+                    .SetToken(SonarQubeApiKey)
+                    .EnableQualityGateWait();
+
+                if (!GitHubActions.IsPullRequest)
+                {
+                    return settings;
+                }
+
+                string sourceBranch = GitHubActions.RefName;
+
+                Information("Analyzing PullRequest {PrId} from branch {BranchName}", GitHubActions.PullRequestNumber, sourceBranch);
+
+                settings = settings
+                    .SetPullRequestKey(GitHubActions.PullRequestNumber.ToString())
+                    .SetPullRequestBranch(sourceBranch);
+
+                return settings;
+            });
+        });
+
+    Target SonarEnd => _ => _
+        .After(UnitTests)
```

Comment:
AFAICT (from the docs [here](https://docs.sonarsource.com/sonarqube-server/10.8/analyzing-source-code/test-coverage/dotnet-test-coverage/#build-process) (the bullet points) we have to run the `SonarEnd` after the coverage report files are generated, so I would for safety change this to:

```suggestion
        .After(CodeCoverage)
```

Or do I miss something?

Reply:
And: in the reports generator README it tells, that a supported _output_ format is SonarQube: https://github.com/danielpalme/ReportGenerator#supported-input-and-output-file-formats

Reply:
> The target CodeCoverage creates HTML report for coverage

Not only the HTML report, also lcov.info (which is needed for coveralls). Don't we need the SQ report files?